.github/workflows/test.yml

```diff
@@ -0,0 +1,66 @@
+name: Test
+on:
+  push:
+    branches: [master]
+  pull_request:
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test:
+    name: ${{ matrix.os }} GHC-${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-24.04]
+        ghc: [9.6.7, 9.8.4, 9.10.1]
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
+      - uses: haskell-actions/setup@v2
+        id: setup-haskell
+        with:
+          ghc-version: ${{ matrix.ghc }}
+      - uses: actions/cache/restore@v4
+        name: Restore cabal store cache
+        id: cache
+        env:
+          # NB: Each `matrix.os` (e.g., `ubuntu-22.04-arm`) uniquely determines
+          # a `runner.arch` (e.g., ARM64), so there is no need to include the
+          # latter as part of the cache key
+          key: cabal-${{ matrix.os }}-ghc${{ matrix.ghc }}
+        with:
+          path: |
+            ${{ steps.setup-haskell.outputs.cabal-store }}
+            dist-newstyle
+          key: |
+            ${{ env.key }}-${{ github.ref }}
+          restore-keys: |
+            ${{ env.key }}-
+      - run: cabal update
```

Comment:
`haskell-actions/setup` performs `cabal update` by default, so no need to do it a second time here.